docker-wrappers/SPRAS/snakemake_long.py

```diff
@@ -0,0 +1,134 @@
+    parser = argparse.ArgumentParser(description="A tool for long-running Snakemake jobs with HTCondor.")
+    # We add a special command to trigger allowing this script to execute the long-running Snakemake command.
+    if isLocal:
+        parser.add_argument("command", help="Helper command to run", choices=["long"])
+    parser.add_argument("--snakefile", help="The Snakefile to run.", required=False)
+    parser.add_argument("--profile", help="A path to a directory containing the desired Snakemake profile.", required=True)
+    # I'd love to change this to "logdir", but using the same name as Snakemake for consistency of feeling between this script
+    # and Snakemake proper.
+    parser.add_argument("--htcondor-jobdir", help="The directory Snakemake will write logs to.", required=False)
+    return parser.parse_args()
+
+"""
+Given a Snakefile, profile, and HTCondor job directory, submit a local universe job that runs
+Snakemake from the context of the submission directory.
+"""
+def submitLocal(snakefile, profile, htcondor_jobdir):
+    # Get the location of this script, which also serves as the executable for the condor job.
+    script_location = pathlib.Path(__file__).resolve()
+
+    submit_description = htcondor.Submit({
+        "executable":              script_location,
+        "arguments":               f"long --snakefile {snakefile} --profile {profile} --htcondor-jobdir {htcondor_jobdir}",
+        "universe":                "local",
+        "request_disk":            "512MB",
+        "request_cpus":            1,
+        "request_memory":          512,
+
+        # Set up logging
+        "log":                     f"{htcondor_jobdir}/snakemake-long.log",
+        "output":                  f"{htcondor_jobdir}/snakemake-long.out",
+        "error":                   f"{htcondor_jobdir}/snakemake-long.err",
+
+        # Specify `getenv` so that our script uses the appropriate environment
+        # when it runs in local universe. This allows the job to access
+        # modules we've installed in the submission environment (notably spras).
+        "getenv":                  "true",
+
+        "JobBatchName":            f"spras-long-{time.strftime('%Y%m%d-%H%M%S')}",
+    })
+
+    schedd = htcondor.Schedd()
+    submit_result = schedd.submit(submit_description)
+
+    print("Snakemake management job was submitted with JobID %d.0. Logs can be found in %s" % (submit_result.cluster(), htcondor_jobdir))
+
+"""
+The top level function for the script that handles file creation/validation and triggers submission of the
+wrapper job.
+"""
+def topMain():
+    args = parseArgs()
+
+    # Check if the snakefile is provided. If not, assume it's in the current directory.
+    if args.snakefile is None:
+        cwd = os.getcwd()
+        args.snakefile = pathlib.Path(cwd) / "Snakefile"
+    if not os.path.exists(args.snakefile):
+        print(f"Error: The Snakefile {args.snakefile} does not exist.")
+        return 1
```

Comment:
This approach of returning error codes is a bit atypical to me. Is this better than raising errors?

Reply:
This is mostly to play better with HTCondor, which expects job executables to return 0 for success, and something else for failure. I guess raising an error _also_ returns non-zero, but this felt more explicit to me. I'll let you make the call.